lib/request-handler.js

```diff
@@ -78,8 +77,8 @@ RequestHandler.prototype.iterateThroughHosts = function (iterator, callback) {
       return callback(new errors.NoHostAvailableError(this.triedHosts));
     }
     host = item.value;
-    // set the distance relative to the client first, using the load balancing policy
-    var distance = host.getDistance(this.loadBalancingPolicy);
+    // set the distance relative to the client first
+    var distance = this.profileManager.getDistance(host);
```

Comment:
But if a user is giving a specific profile as part of their query execution, shouldn't we obey the distance rules for the policy for that profile?

For example, if a user is executing a query that they want to target an analytics or search datacenter, they would use a DCAwareRoundRobinPolicy with the target DC in the profile for those requests, and use a different profile for other requests where they don't want to target those DCs.